hs_core/hydroshare/resource.py

```diff
@@ -34,31 +35,63 @@
     """
     Query iRODS AVU to get quota usage for a user reported in iRODS quota microservices
     :param username: the user name to get quota usage for.
+    :param raise_on_error: if True, raise ValidationError if quota usage cannot be retrieved from iRODS
+    :param zone: the zone to get quota usage for. If None, get quota usage for both user zone and data zone
     :return: the quota usage from iRODS data zone and user zone; raise ValidationError
     if quota usage cannot be retrieved from iRODS
     """
-    attname = username + '-usage'
-    istorage = IrodsStorage()
-    # get quota size for user in iRODS data zone by retrieving AVU set on irods bagit path
-    # collection
+    uqDataZoneSize = get_data_zone_usage(username, raise_on_error=raise_on_error)
+
+    uqUserZoneSize = get_user_zone_usage(username)
+    if uqUserZoneSize < 0:
+        uqUserZoneSize = 0
+        err_msg = f'no quota size AVU for user {username} in userzone'
+        logger.error(err_msg)
+        if raise_on_error:
+            raise ValidationError(err_msg)
+    return uqUserZoneSize, uqDataZoneSize
+
+
+def get_data_zone_usage(username, raise_on_error=True, include_published=False):
+    """
+    Calculate the data zone usage for a given user.
+
+    Args:
+        username (str): The username of the user.
+        raise_on_error (bool, optional): Whether to raise an error if the user does not exist. Defaults to True.
+        include_published (bool, optional): Whether to include published resources in the calculation. Default False.
+
+    Returns:
+        int: The data zone usage in bytes.
+
+    Raises:
+        ValidationError: If the user does not exist and `raise_on_error` is True.
+    """
     try:
-        uqDataZoneSize = istorage.getAVU(settings.IRODS_BAGIT_PATH, attname)
-        if uqDataZoneSize is None:
-            # user may not have resources in data zone, so corresponding quota size AVU may not
-            # exist for this user
-            uqDataZoneSize = -1
-        else:
-            uqDataZoneSize = float(uqDataZoneSize)
-    except SessionException:
-        # user may not have resources in data zone, so corresponding quota size AVU may not exist
-        # for this user
-        uqDataZoneSize = -1
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        logger.error(f'User {username} does not exist')
+        if raise_on_error:
+            raise ValidationError(f'User {username} does not exist')
+        return 0
+    uqDataZoneSize = 0
+    qh_resources = BaseResource.objects.filter(quota_holder=user)
+    if not include_published:
+        qh_resources = qh_resources.exclude(raccess__published=True)
+    qh_resource_ids = qh_resources.values_list('id', flat=True)
+    uqDataZoneSize = ResourceFile.objects.filter(object_id__in=qh_resource_ids).aggregate(Sum('_size'))["_size__sum"]
 
+    return uqDataZoneSize if uqDataZoneSize is not None else 0
+
+
+def get_user_zone_usage(username):
```

Comment:
good catch thank you!
In this PR we get rid of iRods AVUs for the Datazone.
Similar to you, I wish we would get rid of the AVUs for the UserZone as well.
It would certainly be possible, but since the userzone is (:fingers-crossed) going away after our cloud migration, I decided not to invest that effort at this time.